### PR TITLE
Draft revision of README.md

### PR DESCRIPTION
**Pull request by perrydv**
In reviewing our GitHub site, I thought the README.md could achieve a broader appeal and convey more of what NIMBLE is about.  I also wanted to clarify the installation requirements section.  This PR includes a draft revision of README.md for these goals.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,17 +10,43 @@
...
+(aka sequential Monte Carlo), and other general methods.
+
+For writing statistical models, NIMBLE adopts and extends the BUGS
+language, making it largely compatible with
+[BUGS](https://www.mrc-bsu.cam.ac.uk/software/bugs) and
+[JAGS](http://mcmc-jags.sourceforge.net/).  NIMBLE makes BUGS
+extensible, allowing users to add new functions and new distributions.
+
+For writing algorithms (aka analysis methods), NIMBLE provides a
+model-generic programming system, based on two-stage evaluation,
```
Comment (fritzo) [inline README.md]: I'm unclear on the meaning of "based on two-stage evaluation"

**Comment by fritzo** [comment]
:+1: It's great to have more info here

File: README.md (hunk 1 above)
Comment (perrydv) [inline README.md]: It's a term from computer science.  But I removed it.  If it doesn't make sense to you, it is certainly too jargony.

**Comment by paciorek** [comment]
"to rapidly develop hierarchical Bayesian models for statistical analysis" doesn't highlight that we're about algorithm development. But I'm in a talk right now and don't have a specific modification to suggest.

**Comment by perrydv** [comment]
@paciorek I think you are reading the *old* text that I propose to remove.

**Comment by paciorek** [comment]
Indeed. Sorry for the static; new text looks good.

**Pull request merged**